packages/transaction-controller/src/utils/layer1-gas-fee-flow.test.ts

```diff
@@ -0,0 +1,43 @@
+import { CHAIN_IDS } from '../constants';
+import type { Layer1GasFeeFlow, TransactionMeta } from '../types';
+import { getLayer1GasFeeFlow } from './layer1-gas-fee-flow';
+
+const MOCK_UNSUPPORTED_CHAIN_ID = '0xunsupported';
+
+describe('getLayer1GasFeeFlow', () => {
+  const mockLayer1GasFeeFlows = [
+    {
+      matchesTransaction: (transactionMeta: TransactionMeta) =>
+        transactionMeta.chainId === CHAIN_IDS.OPTIMISM,
+    },
+    {
+      matchesTransaction: (transactionMeta: TransactionMeta) =>
+        transactionMeta.chainId === CHAIN_IDS.BASE,
+    },
+  ] as Layer1GasFeeFlow[];
+
+  it('should return the correct Layer1GasFeeFlow based on the transactionMeta', () => {
+    const transactionMetaOptimism = {
```

Comment:
To avoid coupling our tests to a specific implementation, should we avoid referencing `Optimism`?

Also as we use this object multiple times, does it warrant a `TRANSACTION_META_MOCK` like in `gas-flows.test.ts`?

Reply:
Good point Matthew, a similar function added to new tests in https://github.com/MetaMask/core/pull/3944/commits/fa567b3f7eebe1b897bd615c912cb32c6de4801f